src/controllers/sse.ts

```diff
@@ -0,0 +1,205 @@
+import { type Request, type Response, Router } from "express"
+import { instance as createLogger } from "../utils/logger"
+import { validate } from "../validation/sse"
+import { ValidatedRequest } from "../types/express"
+import { envs } from "../config/env"
+import type { BroadcastBody, ConnectionIdParams, HeartbeatEvent, NotificationEvent, SSEConnections, SSEServerEvent } from "../types/sse"
+
+const router = Router()
+const logger = createLogger("routes.sse")
+
+const connections: SSEConnections = new Map<string, Response>()
+const heartbeatInterval = envs.HEARTBEAT_INTERVAL_MS
+
+const generateConnectionId = (): string => {
+  return `conn_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`
+}
+
+export function getConnection(req: Request, res: Response): void {
+  try {
+    res.writeHead(200, {
+      "Content-Type": "text/event-stream",
+      "Cache-Control": "no-cache",
+      "Connection": "keep-alive",
+      "Access-Control-Allow-Origin": "*",
+      "Access-Control-Allow-Headers": "Cache-Control"
+    })
+
+    const connectionId = generateConnectionId()
+    connections.set(connectionId, res)
+
+    const connectedEvent: SSEServerEvent = {
+      type: "connected",
+      connectionId,
+      message: "SSE connection established",
+      timestamp: new Date().toISOString()
+    }
+
+    res.write(`data: ${JSON.stringify(connectedEvent)}\n\n`)
+
+    req.on("close", () => {
+      connections.delete(connectionId)
+      try { 
+        clearInterval(heartbeat) 
+      } catch (error) {
+        logger.error("Error clearing heartbeat interval", { connectionId, error })
+      }
+      logger.info("Client disconnected", { connectionId })
+    })
+
+    const heartbeat = setInterval(() => {
+      if (connections.has(connectionId)) {
+        const heartbeatEvent: HeartbeatEvent = {
+          type: "heartbeat",
+          timestamp: new Date().toISOString()
+        }
+        res.write(`data: ${JSON.stringify(heartbeatEvent)}\n\n`)
+        logger.debug("Heartbeat sent to client", { connectionId })
+      } else {
+        logger.info("Connection no longer exists, clearing heartbeat interval", { connectionId })
+        clearInterval(heartbeat)
+      }
+    }, heartbeatInterval)
+
+    logger.info("SSE connection established", { connectionId })
+  } catch (error) {
+    logger.error("Error establishing SSE connection", { error })
+    res.status(500).json({ error: { message: "Failed to establish SSE connection" } })
+  }
+}
+
+export function sendBroadcast(req: ValidatedRequest<BroadcastBody>, res: Response): void {
+  try {
+    if (!req.input) {
```

Comment:
This should not happen. The validator should have already checked for the object.
All HTTP Codes 400 should be generated by validators.